src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs

```diff
@@ -31,6 +31,10 @@ namespace System.Net.Http
     /// <summary>Provides a set of connection pools, each for its own endpoint.</summary>
     internal sealed class HttpConnectionPoolManager : IDisposable
     {
+        // Used by HttpTelemetry to calculate statistics over all connection pools
+        private static readonly ConcurrentDictionary<HttpConnectionPoolManager, object?> s_allConnectionPools =
+            new ConcurrentDictionary<HttpConnectionPoolManager, object?>();
```

Comment:
Unfortunately to implement this, we have to keep track of all connection pools, even if telemetry is disabled (since we can't do it after the fact). Is there an alternative approach where we can avoid this?

Reply:
Also, since this is effectively a `ConcurrentHashSet`, is it more efficient to use `object` or something like `byte` for the dummy value?

Reply:
This is a potentually huge leak. Pools are now rooted until they're disposed.

Reply:
Changed to keep `WeakReference`s around instead, and use a finalizer to ensure the pool manager is removed from the global queue.